src/wms/src/main/java/applicationContext.xml

```diff
@@ -408,10 +414,12 @@
 	</bean>
 
     <!-- cite compliance hack -->
-    <bean id="wmsCiteComplianceHack" class="org.geoserver.ows.CiteComplianceHack">
-        <constructor-arg ref="geoServer"/>
-        <constructor-arg value="org.geoserver.wms.WMSInfo"/>
-    </bean>
+    <mvc:interceptors>
+        <bean id="wmsCiteComplianceHack" class="org.geoserver.ows.CiteComplianceHack">
+            <constructor-arg ref="geoServer"/>
+            <constructor-arg value="org.geoserver.wms.WMSInfo"/>
+        </bean>
+    </mvc:interceptors>
```

Comment:
I heard that in order to make WMS compliant, one had to make all the other services compliant... but this change seems to be related to having the interceptor be actually invoked.  The same interceptor is declared for other services as well, why has it been changed only here?

I've done a quick test and here are some thoughts.

The situation is confusing. Each OGC service should have its own compliance hack declaration, but only a few actually do, WMS, WFS, and CSW (but not WCS for example, maybe it's strict by default):

```xml
C:\Users\andre\devel\geoserver>git grep -i citecompliancehack -- "*.xml"
src/extension/csw/core/src/main/java/applicationContext.xml:          <ref bean="cswCiteComplianceHack"/>
src/extension/csw/core/src/main/java/applicationContext.xml:    <bean id="cswCiteComplianceHack" class="org.geoserver.ows.CiteComplianceHack">
src/wfs/src/main/java/applicationContext.xml:   <bean id="wfsCiteComplianceHack" class="org.geoserver.ows.CiteComplianceHack">
src/wfs/src/main/java/applicationContext.xml:                           <ref bean="wfsCiteComplianceHack"/>
src/wms/src/main/java/applicationContext.xml:    <bean id="wmsCiteComplianceHack" class="org.geoserver.ows.CiteComplianceHack">
```

Originally the all used to run, maybe some Spring upgrade broke that? In any case, looking at the compliance hack source code, it makes no attempt to check which service is being called. So if all 3 of the above hacks are called, the last one wins, which is a random choice picking the cite compliance configuration of the last hack that gets run.

The hack should be doing some attempt at figuring out if the service is a match, e.g., check the last element of the path, and/or check if there is a service=<id> kvp parameter that matches the current service configuration.

My other worry is that the mvc configuration is something new, and it might end up interacting with the other dispatchers, like the REST API and the OGC API one... hopefully not, the cite compliance hack specifically checks that the handler is the OWS Dispatcher. But I'd recommend doing some light manual testing of the REST API and the OGC API in a running GeoServer.

Long story short:
* Update the hack to check its own configured service only
* Update other OGC services with CITE hacks as well
* Add a test to ensure the flag is working, it's evidently missing or we would not have had this regression

Reply:
Yessir!

Reply:
@aaime turns out that there's no need to introduce the `mvc:interceptors` tag. There was just a missing initialized element on the `wmsURLMapping` bean.

I've been added few more tests and checks for the current service.